src/transformers/trainer.py

```diff
@@ -1462,7 +1462,10 @@ def _load_state_dict_in_model(self, state_dict):
     def _maybe_log_save_evaluate(self, tr_loss, model, trial, epoch, ignore_keys_for_eval):
         if self.control.should_log:
             logs: Dict[str, float] = {}
-            tr_loss_scalar = tr_loss.item()
+
+            # all_gather + mean() to get average loss over all processes
+            tr_loss_scalar = self._nested_gather(tr_loss).mean().item()
```

Comment:
Very clean!
Is it properly working without putting it in a 1d tensor? It's always tricky with gather.

Reply:
I ran a dummy training script for distributed training and the training logs were displayed, so I think it works correctly for a scalar tensor. 

Maybe you can run this script in a distributed setup - e.g. Vorace wiht 4 GPUs - to make sure everything works as expected:

```bash
python -m torch.distributed.launch \
        --nproc_per_node 4 run_speech_recognition_ctc.py \
        --dataset_name="common_voice" \
        --model_name_or_path="facebook/wav2vec2-base" \
        --dataset_config_name="ab" \
        --output_dir="./dummy" \
        --preprocessing_num_workers="1" \
        --overwrite_output_dir \
        --num_train_epochs="2" \
        --per_device_train_batch_size="1" \
        --gradient_accumulation_steps="1" \
        --learning_rate="3e-4" \
        --warmup_steps="500" \
        --evaluation_strategy="steps" \
        --audio_column_name="path" \
        --text_column_name="sentence" \
        --save_steps="400" \
        --eval_steps="100" \
        --logging_steps="1" \
        --layerdrop="0.0" \
        --save_total_limit="3" \
        --freeze_feature_extractor \
        --chars_to_ignore , ? . ! - \; \: \" “ % ‘ ” � \
        --fp16 \
        --group_by_length \
        --do_train --do_eval
```

Reply:
with `run_speech_recognition_ctc.py` being here: https://github.com/huggingface/transformers/blob/master/examples/pytorch/speech-recognition/run_speech_recognition_ctc.py

Reply:
Tested locally and it works perfectly indeed. Thanks!